include/blas/device.hh

```diff
@@ -369,32 +369,43 @@ inline const char* device_error_string( rocblas_status error )
 
 // -----------------------------------------------------------------------------
 // set/get device functions
+[[deprecated("use blas::Queues& with all blaspp calls")]]
 void set_device( blas::Device device );
+
+// private, internal routine; sets device for cuda, rocm; nothing for onemkl
+void internal_set_device( blas::Device device );
+
+[[deprecated("use blas::Queues& with all blaspp calls")]]
 void get_device( blas::Device *device );
+
 device_blas_int get_device_count();
 #ifdef BLAS_HAVE_ONEMKL
-void enumerate_devices(std::vector<cl::sycl::device> &devices);
+void enumerate_devices(std::vector<sycl::device> &devices);
 #endif
 
 // -----------------------------------------------------------------------------
 // memory functions
 
 /// @deprecated: use device_free( ptr, queue ).
+[[deprecated("use device_free( ptr, queue )")]]
 void device_free( void* ptr );
 
 void device_free( void* ptr, blas::Queue &queue );
 
 /// @deprecated: use host_free_pinned( ptr, queue ).
+// todo: does this really need to be deprecated
```

Comment:
I would say yes, because it will throw blas::Error( "unsupported function for sycl backend", __func__ );
https://github.com/icl-utk-edu/blaspp/blob/master/src/device_utils.cc#L183

Reply:
As discussed, we will do this in a separate PR.